bin/generate-tests.php

```diff
@@ -73,27 +73,29 @@
     }
 
     $class = u($command['name'])->camel()->title()->append('Test')->toString();
-    $task = $command['name'];
-    $methodName = u($command['name'])->replace(':', '_')->replace('-', '_')->camel()->title()->toString();
 
-    add_test($args, $class, $task, $methodName);
+    add_test($args, $class);
 }
 
-add_test(['context:context', '--context', 'run'], 'ContextContextRun', 'context:context');
-add_test(['context:context', '--context', 'my_default', '-vvv'], 'ContextContextMyDefault', 'context:context');
-add_test(['context:context', '--context', 'no_no_exist'], 'ContextContextDoNotExist', 'context:context');
-add_test(['context:context', '--context', 'production'], 'ContextContextProduction', 'context:context');
+add_test(['context:context', '--context', 'run'], 'ContextContextRunTest');
+add_test(['context:context', '--context', 'my_default', '-vvv'], 'ContextContextMyDefaultTest');
+add_test(['context:context', '--context', 'no_no_exist'], 'ContextContextDoNotExistTest');
+add_test(['context:context', '--context', 'production'], 'ContextContextProductionTest');
 add_test(['parallel:sleep', '--sleep5', '0', '--sleep7', '0', '--sleep10', '0'], 'ParallelSleepTest');
+add_test([], 'NewProjectTest', '/tmp');
+add_test(['unknown:command', 'toto', '--foo', 1], 'NoConfigTest', '/tmp');
 
-function add_test(array $args, string $class)
+function add_test(array $args, string $class, string $cwd = null)
 {
     $fp = fopen(__FILE__, 'r');
     fseek($fp, __COMPILER_HALT_OFFSET__ + 1);
     $template = stream_get_contents($fp);
 
+    $bin = __DIR__ . '/castor';
+
     $process = new Process(
-        [\PHP_BINARY, 'bin/castor', ...$args],
-        cwd: __DIR__ . '/../',
+        [\PHP_BINARY, $bin, ...$args],
+        cwd: $cwd ?: __DIR__ . '/../',
```

Comment:
could you move the default value to the function argument?

Reply:
Nope, because it's easier to check if a custom cwd was passed or not, so the argument is not specified in the generated test case